sorting.go

```diff
@@ -25,6 +25,11 @@ func (p peerSorterArr) Less(a, b int) bool {
 //
 
 func copyPeersFromList(target ID, peerArr peerSorterArr, peerList *list.List) peerSorterArr {
+	if cap(peerArr) < len(peerArr)+peerList.Len() {
+		newArr := make(peerSorterArr, 0, len(peerArr)+peerList.Len())
```

Comment:
Is the `peerArr` being reused frequently/multiple times? If it is then we could reduce allocations further if we over-allocate slightly here. Either to power of two or by some fixed margin.

Reply:
Actually, we can allocate entirely up-front (we can pre-compute the size). However, as-is, we're only going to have 3 allocations max.

Reply:
That is, we call this function at most three times (ok, technically 4).